client/post-editor/editor-action-bar/index.jsx

```diff
@@ -14,28 +15,30 @@ import Tooltip from 'components/tooltip';
 import Button from 'components/button';
 import EditorActionBarViewLabel from './view-label';
 import EditorStatusLabel from 'post-editor/editor-status-label';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPost } from 'state/posts/selectors';
+import PostEditStore from 'lib/posts/post-edit-store';
 
-export default React.createClass( {
+class EditorActionBar extends Component {
 
-	displayName: 'EditorActionBar',
-
-	propTypes: {
+	static propTypes = {
 		isNew: React.PropTypes.bool,
 		onPrivatePublish: React.PropTypes.func,
 		post: React.PropTypes.object,
 		savedPost: React.PropTypes.object,
 		site: React.PropTypes.object,
 		type: React.PropTypes.string
-	},
+	};
 
-	getInitialState: function() {
-		return {
-			viewLinkTooltip: false
-		};
-	},
+	state = {
+		viewLinkTooltip: false
+	};
 
 	render() {
 		const multiUserSite = this.props.site && ! this.props.site.single_user_site;
+		const isPasswordProtected = utils.getVisibility( this.props.post ) === 'password';
```

Comment:
@youknowriad It does indeed feel weird :( It looks like we're storing password changes in Redux though ([example](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/editor-visibility/index.jsx#L180)) so I think we need to stick with `this.props.post`. Changing to the Flux post causes some unexpected behavior when you change from password -> public -> password.